momepy/functional/_diversity.py

```diff
@@ -48,72 +138,103 @@ def describe(
     DataFrame
         A DataFrame with descriptive statistics.
     """
-    try:
-        from numba import njit
-    except (ModuleNotFoundError, ImportError):
-        warnings.warn(
-            "The numba package is used extensively in this function to accelerate the"
-            " computation of statistics but it is not installed or  cannot be imported."
-            " Without numba, these computations may become slow on large data.",
-            UserWarning,
-            stacklevel=2,
-        )
-        from libpysal.common import jit as njit
-
-    @njit
-    def _mode(array):
-        """Custom mode function for numba."""
-        array = np.sort(array.ravel())
-        mask = np.empty(array.shape, dtype=np.bool_)
-        mask[:1] = True
-        mask[1:] = array[1:] != array[:-1]
-        unique = array[mask]
-        idx = np.nonzero(mask)[0]
-        idx = np.append(idx, mask.size)
-        counts = np.diff(idx)
-        return unique[np.argmax(counts)]
-
-    @njit
-    def _describe(values, q, include_mode=False):
-        """Helper function to calculate average."""
-        nan_tracker = np.isnan(values)
-
-        if (len(values) > 2) and (not nan_tracker.all()):
-            if nan_tracker.any():
-                lower, higher = np.nanpercentile(values, q)
-            else:
-                lower, higher = np.percentile(values, q)
-            values = values[(lower <= values) & (values <= higher)]
-
-        results = [
-            np.mean(values),
-            np.median(values),
-            np.std(values),
-            np.min(values),
-            np.max(values),
-            np.sum(values),
-        ]
-        if include_mode:
-            results.append(_mode(values))
-        return results
-
     if not isinstance(y, Series):
         y = Series(y)
 
     grouper = y.take(graph._adjacency.index.codes[1]).groupby(
         graph._adjacency.index.codes[0]
     )
 
-    if q is None:
-        stat_ = grouper.agg(["mean", "median", "std", "min", "max", "sum"])
-        if include_mode:
-            stat_["mode"] = grouper.agg(lambda x: _mode(x.values))
+    return _compute_stats(grouper, q, include_mode)
+
+
+def describe_reached(
+    y, y_group, result_index=None, graph=None, q=None, include_mode=False
+) -> DataFrame:
+    """
+    Calculates statistics of ``y`` objects reached on a street network.
```

Comment:
Not only street network. Graph or IDs can come from anywhere. You can easily use enclosure ID here instead. The function is pretty generic.